plugins/node/opentelemetry-instrumentation-pg/test/pg-pool.test.ts

```diff
@@ -482,4 +509,91 @@ describe('pg-pool', () => {
       });
     });
   });
+
+  describe('pg metrics', () => {
+    let otelTestingMeterProvider;
+    let inMemoryMetricsExporter: InMemoryMetricExporter;
+
+    function initMeterProvider() {
+      otelTestingMeterProvider = new MeterProvider();
+      inMemoryMetricsExporter = new InMemoryMetricExporter(
+        AggregationTemporality.CUMULATIVE
+      );
+      const metricReader = new PeriodicExportingMetricReader({
+        exporter: inMemoryMetricsExporter,
+        exportIntervalMillis: 100,
+        exportTimeoutMillis: 100,
+      });
+
+      otelTestingMeterProvider.addMetricReader(metricReader);
+      instrumentation.setMeterProvider(otelTestingMeterProvider);
+    }
+
+    beforeEach(() => {
+      initMeterProvider();
+      inMemoryMetricsExporter.reset();
+    });
+
+    it('should generate `db.client.connection.count` and `db.client.connection.pending_requests` metrics', async () => {
+      const span = provider.getTracer('test-pg-pool').startSpan('test span');
+      context.with(trace.setSpan(context.active(), span), () => {
+        pool.connect((err, client, release) => {
+          if (err) {
+            throw new Error(err.message);
+          }
+          if (!release) {
+            throw new Error('Did not receive release function');
+          }
+          if (!client) {
+            throw new Error('No client received');
+          }
+          assert.ok(client);
+
+          client.query('SELECT NOW()', async (err, ret) => {
+            release();
+            if (err) {
+              throw new Error(err.message);
+            }
+            assert.ok(ret);
+
+            const exportedMetrics = await waitForNumberOfExports(
+              inMemoryMetricsExporter,
+              2
+            );
+            const metrics = exportedMetrics[1].scopeMetrics[0].metrics;
+
+            assert.strictEqual(
+              metrics[0].descriptor.name,
+              'db.client.connection.count'
+            );
+            assert.strictEqual(
+              metrics[0].descriptor.description,
+              'The number of connections that are currently in state described by the state attribute.'
+            );
+            assert.strictEqual(
+              metrics[0].dataPoints[0].attributes[
+                SEMATTRS_CLIENT_CONNECTION_STATE
+              ],
+              'used'
+            );
+            assert.strictEqual(
+              metrics[0].dataPoints[1].attributes[
+                SEMATTRS_CLIENT_CONNECTION_STATE
+              ],
+              'idle'
+            );
+
+            assert.strictEqual(
+              metrics[1].descriptor.name,
+              'db.client.connection.pending_requests'
+            );
+            assert.strictEqual(
+              metrics[1].descriptor.description,
+              'The number of current pending requests for an open connection.'
```

Comment:
is it on purpose that this test never asserts on the actual data point values? :thinking:

Reply:
@pichlermarc  Not on purpose. I added the check on the datapoints now, but I'm having an issue. When I tested manually, everything was working as expected, but now this test is failing because it says the data point for used is `-1`, instead of the correct `1`. I printed the values getting added to the metric and it has all the right values, adding the correct 1 when it starts and removing it when the connection is no longer being used.

The data point for the used connection is like this, which I noticed has start and end times, could this aggregation be causing the value to change? I can't figure it out what else could be wrong on this. Open to suggestions and ideas
```
{
    attributes: {
      'db.client.connection.state': 'used',
      'db.client.connection.pool.name': 'localhost:54320/postgres'
    },
    startTime: [ 1723575108, 137000000 ],
    endTime: [ 1723575108, 486000000 ],
    value: -1
  },
```

Reply:
After some updates, now the value is 0, instead of -1, but still not the expected 1